src/components/shared/DataTable.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react'
+import Paper from '@mui/material/Paper'
+import Table from '@mui/material/Table'
+import TableBody from '@mui/material/TableBody'
+import TableCell from '@mui/material/TableCell'
+import TableContainer from '@mui/material/TableContainer'
+import TableHead from '@mui/material/TableHead'
+import TablePagination from '@mui/material/TablePagination'
+import TableRow from '@mui/material/TableRow'
+import { Box, FormControlLabel, Switch } from '@mui/material'
+import SearchIcon from '@mui/icons-material/Search'
+import TextField from '@mui/material/TextField'
+import Autocomplete from '@mui/material/Autocomplete'
+import { Search, SearchIconWrapper, StyledInputBase } from '@/utils/searchInputStyles'
+
+interface DataTableProps {
+  searchInput: boolean
+  isSortTextField: boolean
+  sortLabel?: string
+  sortOptionsData?: string[]
+  columnsData: Column[]
+  rowData: any
+}
+
+interface Column {
```

Comment:
move this into a new file and export it

Reply:
Lets also change `id` to `key` and `label` to `title` or `header`

Reply:
Fixed